src/lib/operators/abstract_operator.cpp

```diff
@@ -20,9 +20,21 @@ void AbstractOperator::execute() {
 
   auto transaction_context = this->transaction_context();
 
-  if (transaction_context) transaction_context->on_operator_started();
-  _output = _on_execute(transaction_context);
-  if (transaction_context) transaction_context->on_operator_finished();
+  if (transaction_context) {
```

Comment:
Do we have a convention regarding checking for `nullptr`?

Reply:
Nope. If you think we should have one, feel free to open an issue for it. Personally, I am fine with either and also with seeing them mixed.